src/VecSim/vec_sim_index.h

```diff
@@ -0,0 +1,109 @@
+#pragma once
+#include "vec_sim_common.h"
+#include "query_results.h"
+#include <stddef.h>
+
+/**
+ * @brief Abstract C++ class for vector index, delete and lookup
+ *
+ */
+class VecSimIndex {
+protected:
+    size_t dim;
+    VecSimType vecType;
+    VecSimMetric metric;
```

Comment:
Why are they `protected` and not `private`? You have getters to access these fields

Reply:
for convenience only

Reply:
Don't you think that it is better use `private` instead? So we won't bread abstraction